### ISSUE TITLE
Be ready to use after installation/enabling without restart

### ISSUE THREAD

**Issue by gerardroche** [SublimeLinter/SublimeLinter#1397]
When enabling SL via Package Control I get an exception:

```
Traceback (most recent call last):
  File "/home/code/sublime_text_3/sublime_plugin.py", line 183, in reload_plugin
    m.plugin_loaded()
  File "/home/code/.config/sublime-text-3/Packages/SublimeLinter/sublime_linter.py", line 48, in plugin_loaded
    style.read_gutter_theme()
  File "/home/code/.config/sublime-text-3/Packages/SublimeLinter/lint/style.py", line 23, in read_gutter_theme
    theme_file = os.path.basename(theme_path)
  File "./python3.3/posixpath.py", line 142, in basename
AttributeError: 'NoneType' object has no attribute 'rfind'
```

Ubuntu, ST 3173, SL 4.5.1

Reproduce: 

* Disable SL via PC
* Restart ST
* Enable SL via PC

**Comment by kaste** [comment]
Check against None? => #1398

**Comment by kaste** [comment]
So it seems that after just enabling the package, the settings all return None. They just don't load or reload. If I fix this None check it throws somewhere else and basically SL is unusable.

This is not even a race. I checked the settings API manually. So if this is gets a feat request to allow enabling/disabling the whole SL without the need for a restart, I am first a bit clueless and need to dig deeper.

**Comment by kaste** [comment]
@gerardroche It should be ready without restart after install/upgrade, just disable/enable dance doesn't work.

**Comment by kaste** [comment]
Scenario: Startup ST with disabled SL. Then enable via PC.

- With git check out. ST will not load the default settings. Triggering `os.utime('.../SublimeLinter/SublimeLinter.sublime-settings')` will not reload these settings. 👎 

- With PC install. All plugins would have thrown already bc they cannot import SublimeLinter on startup if SL is disabled. ST will correctly load the default settings. But then our reloader cannot reload and register the plugins bc ST does not put them inside `sys.modules` if they throw at import time.  👎

**Comment by kaste** [comment]
Closing since this didn't move forward. Bc of the interdependence of plugins and SublimeLinter as a framework any solution will be non-trivial.

As I see it, we must intercept 'disabling SL', then also disable all SL plugins and remember what we did. If we now intercept 'enabling SL' we also also enable all SL plugins we disabled before but not the ones the user had disabled before. 

I especially hate the 'remember what we did' thing herein.
